scripts/coverage.sh

```
@@ -0,0 +1,46 @@
+# Get the list of packages to test, excluding those in the ignore list
+IGNORED=$(cat ./scripts/coverage_ignore.txt | xargs -I{} echo '-not -path ./{}/*' | xargs)
+PACKAGES=$(go list ./pkg/... ./internal/... ./cmd/... | grep -v -f ./scripts/coverage_ignore.txt)
+
+echo "${BLUE}Running tests on packages:${NC}"
+echo "$PACKAGES"
+
+# Run the tests and generate coverage profile
+go test -cover $PACKAGES -coverprofile="$TEST_REPORTS_DIR/coverage.out"
+
+# Print coverage summary
+printf "\n${GREEN}Coverage Summary:${NC}\n"
```

Shellcheck (0.11.0)

[info] Don't use variables in the printf format string. Use printf '..%s..' "$foo".

(SC2059)